### PR TITLE
Fixes sending OnSystemRequest with HTTP mode

### PR DESCRIPTION
**Pull request by dev-gh**
PolicyHandler::SendMessageToSDK was using obsolete logic of choosing appropriate application to send OnSystemRequest. Changes replace it with call to GetAppIdForSending() which considers application HMI level, registration status, device consent etc. 
Primary source of changes is https://github.com/ATymchenko/sdl_core/commit/2c8677595c0e85c36871774bb1993a783f0f511f

Also appropriate unit test have been modified.

**Note:** there is a **dependency** on https://github.com/smartdevicelink/sdl_core/pull/1548 so these PRs both have to be merges in order to pass ATF tests for HTTP mode.

### CODE REVIEW

**Comment by dev-gh** [comment]
@AKalinich-Luxoft @LitvinenkoIra Please review

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 1]
```diff
@@ -1021,18 +1021,11 @@ bool PolicyHandler::SendMessageToSDK(const BinaryMessage& pt_string,
...
-  } else {
-    app_id = last_used_app_ids_.back();
+  uint32_t app_id = GetAppIdForSending();
 
-    app = application_manager_.application(app_id);
-  }
+  ApplicationSharedPtr app = application_manager_.application(app_id);
 
-  if (!app) {
+  if (!app.valid()) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/policies/policy_handler.cc]: @dev-gh please use overloaded boolean operator

Comment (dev-gh) [inline src/components/application_manager/src/policies/policy_handler.cc]: @LuxoftAKutsan 
Done with  ce846b9

**Comment by dev-gh** [comment]
@AByzhynar Please review

**Comment by dev-gh** [comment]
@AByzhynar Please review

**Comment by jacobkeeler** [comment]
Closing as duplicate of #1607

**Pull request closed without merging**